### PR TITLE
not showing notification when tab is already opened

### PR DESCRIPTION
**Pull request by dirathea**
Not displaying desktop notification when the url already in one of the opened tab

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Referenced issues

This pull request has been submitted to:
- [#98: Do not show desktop notifications if I have the issue opened in one of my tabs](https://issuehunt.io/repos/18894771/issues/98)
---

IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

### CODE REVIEW

**Comment by dirathea** [comment]
try to solve issue #98

**Comment by dirathea** [comment]
@notlmn could you check this one out?

File: source/lib/notifications-service.js [hunk 1]
```diff
@@ -103,15 +103,27 @@ export function getNotificationObject(notificationInfo) {
...
+	await browser.notifications.create(notificationId, notificationObject);
+	await localStore.set(notificationId, notification);
+}
 
-		await delay(50);
+export async function showNotifications(notifications) {
+	for (const notification of notifications) {
+		try {
+			const urlToOpen = await notificationHandlers[notification.subject.type](notification);
+			const tabs = await queryTabs([urlToOpen]);
```
Comment (notlmn) [inline source/lib/notifications-service.js:118]: `queryTabs` fails if the user has not granted the `tabs` permission, which we currently request for the "Reuse tabs" option.

If we assume that this permission might not be granted, then additional checks need to be applied here and also end up making this an option to request that permission (take a look at https://github.com/sindresorhus/notifier-for-github/pull/161, https://github.com/sindresorhus/notifier-for-github/pull/162, and https://github.com/sindresorhus/notifier-for-github/commit/a7af74c2adada9b73fc786649a903fbef548bd01 for context on asking permissions).

Both of the options above, I'm opposed to adding right now. And `api.openTab` already switches to a tab if we have one already open.

Thanks for the PR, but showing a notification with the existing behavior seems logical than not showing one. I'm gonna pass out on this one.

**Comment by dirathea** [comment]
Ah that's sounds logical, ok I get it.
Maybe this issue also shouldn't had bounty I guess.

**Pull request closed without merging**